presto-main/src/main/java/com/facebook/presto/server/RequestErrorTracker.java

```diff
@@ -116,7 +116,9 @@ public void requestFailed(Throwable reason)
         }
 
         if (reason instanceof RejectedExecutionException) {
-            throw new PrestoException(errorCode, reason);
+            if (!reason.getMessage().contains("Max requests queued per destination")) {
```

Comment:
Add a comment explaining the rationale?  I.e. we want to do exponential backoff to allow the OOT killer to kill queries and not fail outright